src/app/pages/signing/signing.component.ts

```diff
@@ -1,65 +1,91 @@
+   * @param textbox - textbox in which user types.
+   * @param formatCharacters - formatting character(s) to be placed around the selected text
+   */
+  addFormatting(textbox: HTMLTextAreaElement, formatCharacters: string) {
     const start = textbox.selectionStart;
     const end = textbox.selectionEnd;
     const text = textbox.value;
-    this.signingBoxText = text.slice(0, start) + '**' + text.slice(start, end) + '**' + text.slice(end);
+    this.signingBoxText = text.slice(0, start) + formatCharacters + text.slice(start, end) + formatCharacters + text.slice(end);
     textbox.select();
   }
 
-  makeItalics(textbox: HTMLTextAreaElement) {
-    const start = textbox.selectionStart;
-    const end = textbox.selectionEnd;
-    const text = textbox.value;
-    this.signingBoxText = text.slice(0, start) + '*' + text.slice(start, end) + '*' + text.slice(end);
-    textbox.select();
+  /**
+   * Update character count value and text colour
+   * @param textbox - textbox in which user types.
+   */
+  updateCount(textbox: HTMLTextAreaElement) {
+    this.charCount = textbox.value.length;
+    this.charCountColor = (this.charCount > this.maxCharCount ) ? '#EE1111' : '#b0b0b0';
+    this.canSend = (0 < this.charCount && this.charCount <= this.maxCharCount) ? true : false;
   }
 
-  makeUnderline(textbox: HTMLTextAreaElement) {
-    const start = textbox.selectionStart;
-    const end = textbox.selectionEnd;
-    const text = textbox.value;
-    this.signingBoxText = text.slice(0, start) + '<u>' + text.slice(start, end) + '</u>' + text.slice(end);
-    textbox.select();
-  }
+  /**
+   * Creates a new blank message and fills in all the info before sending it
+   * @param textbox - the textbox where the contents of the message are retrieved from (not the preview box)
+   */
+  createMsg(textbox: HTMLTextAreaElement) {
+    const newMsg = this.msgIo.createBlankMessage();
+    newMsg.backColor = this.backgroundColor;
+    newMsg.fontColor = this.textColor;
+    newMsg.fontFamily = this.currentFont;
+    newMsg.contents = textbox.value;
+    newMsg.to = this.uid;
 
-  makeStrikethrough(textbox: HTMLTextAreaElement) {
-    const start = textbox.selectionStart;
-    const end = textbox.selectionEnd;
-    const text = textbox.value;
-    this.signingBoxText = text.slice(0, start) + '~~' + text.slice(start, end) + '~~' + text.slice(end);
-    textbox.select();
+    this.msgIo.sendMessage(newMsg, this.key).then(() =>
+      this.router.navigate(['/msg-sent']))
+      .catch(error => alert(error));
   }
 
-  chooseFontColour() {
-    $('#colorpicker-font').trigger('click');
-  }
+  /**
+   * Extract query parameters from ActivatedRoute and maximum message length from MsgIoService
+   */
+  constructor(private route: ActivatedRoute, private msgIo: MsgIoService, private router: Router) {
+    this.route.queryParams.subscribe(params => {
+      this.uid = params['uid'];
+      this.key = params['key'];
```

Comment:
TSLint: Object access via string literals is not allowed here
```suggestion
      this.key = params.key;
```